Ginger/GingerCore/ALM/Qtest/ExportToQtest.cs

```diff
@@ -0,0 +1,594 @@
+            description = description.Replace("<<&Actions&>>", actsDesc);
+            step.StepDescription = description;
+
+            string expectedTemplate =
+               "<html><body><div align=\"left\"><font face=\"Arial\"><span style=\"font-size:8pt\"><<&Expected&&>></span></font></div></body></html>";
+            step.StepExpectedResult = expectedTemplate.Replace("<<&Expected&&>>", activity.Expected);
+
+            step.Post();
+
+            activity.ExternalID = step.ID.ToString();
+            return true;
+        }
+
+        public static bool ExportBusinessFlowToQtest(BusinessFlow businessFlow, TestSet mappedTestSet, string uploadPath, ObservableList<ExternalItemFieldBase> testSetFields, ref string result)
+        {
+            TestSet testSet;
+            ObservableList<ActivitiesGroup> existingActivitiesGroups = new ObservableList<ActivitiesGroup>();
+            try
+            {
+                if (mappedTestSet == null)
+                {
+                    //##create new Test Set in QC
+                    TestSetFactory TestSetF = (TestSetFactory)mTDConn.TestSetFactory;
+                    testSet = (TestSet)TestSetF.AddItem(System.DBNull.Value);                    
+
+                    //set the upload path
+                    TestSetTreeManager treeM = (TestSetTreeManager)mTDConn.TestSetTreeManager;
+                    ISysTreeNode testSetParentFolder = (ISysTreeNode)treeM.get_NodeByPath(uploadPath);
+                    testSet.TestSetFolder = testSetParentFolder.NodeID;
+                }
+                else
+                {
+                    //##update existing test set
+                    //testSet = mappedTestSet;
+                    testSet = ImportFromQC.GetQCTestSet(mappedTestSet.ID.ToString());
+                     
+                    TSTestFactory testsF = (TSTestFactory)testSet.TSTestFactory;
+                    List tsTestsList = testsF.NewList("");
+                    foreach (TSTest tsTest in tsTestsList)
+                    {
+                        ActivitiesGroup ag = businessFlow.ActivitiesGroups.Where(x => (x.ExternalID == tsTest.TestId.ToString() && x.ExternalID2 == tsTest.ID.ToString())).FirstOrDefault();
+                        if (ag == null)
+                            testsF.RemoveItem(tsTest.ID);
+                        else
+                            existingActivitiesGroups.Add(ag);
+                    }
+                }
+                
+                //set item fields
+                foreach (ExternalItemFieldBase field in testSetFields)
+                {
+                    if (field.ToUpdate || field.Mandatory)
+                    {
+                        if (string.IsNullOrEmpty(field.SelectedValue) == false && field.SelectedValue != "NA")
+                            testSet[field.ID] = field.SelectedValue;
+                        else
+                            try { testSet[field.ID] = "NA"; }
+                            catch { }
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Handle the exception or explain in a comment why it can be ignored.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5395796)